src/Idris/Desugar.idr

```diff
@@ -209,21 +209,22 @@ mutual
   desugarB side ps (PLam fc rig p pat argTy scope)
       = pure $ ILam EmptyFC rig !(traverse (desugar AnyExpr ps) p)
                    (Just (MN "lamc" 0)) !(desugarB AnyExpr ps argTy) $
-                 ICase fc (IVar EmptyFC (MN "lamc" 0)) (Implicit fc False)
+                 ICase fc top (IVar EmptyFC (MN "lamc" 0)) (Implicit fc False)
```

Comment:
This should probably `rig` rather than top. Same comment for above.

Reply:
Thanks, should be fixed in the latest commit, with a test case for each.